src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -247,6 +247,10 @@ private static Assembly ResolveForKnownRequestor(string requestedAssemblyName, A
 
                     string directoryPath = Path.GetDirectoryName(requestingAssemblyFullPath);
                     string assemblyFullPath = Path.Combine(directoryPath, assemblyIdentity.Name + ".dll");
+                    if (!File.Exists(assemblyFullPath))
+                    {
+                        return null;
+                    }
 
```

Comment:
I don't understand this. Two questions:
1. If this we weren't checking for File.Exists, then LoadCore would throw an exception - what are the semantics of throwing an exception in the assembly resolve event? How were we catching this exception and logging a warning before? 

2. It sounds like we return null instead of throwing an exception, the CLR continues its search for the fallback - why doesn't this happen in the IDE and happens only in the compiler? Have you found any clues to why this worked fine a period in between and suddenly seems to have regressed?